src/main/java/internet/store/dao/jdbc/ShopCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,141 @@
+            }
+            do {
+                Long productId = resultSet.getObject("id", Long.class);
+                if (productId != null) {
+                    productList.add(retrieveProductDataFromDB(resultSet));
+                }
+            } while (resultSet.next());
+            shoppingCart.setProducts(productList);
+            return Optional.of(shoppingCart);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Shopping cart for user "
+                    + userId + " not found", ex);
+        }
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        String query = "INSERT INTO shopping_cart (user_id) values (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, shoppingCart.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setCartId(resultSet.getLong(1));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Shopping cart "
+                    + shoppingCart + " was not created", ex);
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        deleteProductsFromCart(shoppingCart.getCartId());
+        for (Product product : shoppingCart.getProducts()) {
+            addProductToCart(shoppingCart.getCartId(), product.getId());
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean delete(Long cartId) {
+        String query = "UPDATE shopping_cart SET deleted = true WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, cartId);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Shopping cart for card ID "
+                    + cartId + " was not deleted", ex);
+        }
+    }
+
+    private boolean deleteProductsFromCart(Long cartId) {
+        String queryDelete = "DELETE FROM shopping_cart_product WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryDelete)) {
+            statement.setLong(1, cartId);
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("no products from cart "
+                    + cartId + " was not deleted", ex);
+        }
+        return true;
+    }
+
+    private boolean addProductToCart(Long cartId, Long productId) {
```

Comment:
change this method to addProductsToCart 
and fix it's impl according to 8 point from checklist 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_23.html